modules/shell/command.go

```diff
@@ -73,20 +74,29 @@ func RunCommandAndGetStdOut(t testing.TestingT, command Command) string {
 
 // RunCommandAndGetStdOutE runs a shell command and returns solely its stdout (but not stderr) as a string. The stdout
 // and stderr of that command will also be printed to the stdout and stderr of this Go program to make debugging easier.
+// Any returned error will be of type ErrWithCmdOutput, containing the output streams and the underlying error.
 func RunCommandAndGetStdOutE(t testing.TestingT, command Command) (string, error) {
-	stdout := []string{}
-	stderr := []string{}
-	err := runCommandAndStoreOutputE(t, command, &stdout, &stderr)
+	output, err := runCommand(t, command)
+	if err != nil {
+		return output.stdout.String(), &ErrWithCmdOutput{err, output}
```

Comment:
Should this not do the `strings.Join` thing as a couple lines below?

Reply:
They are equal, but instead of doing `strings.Join` here, I'll replace the one below with `output.stdout.String()`, so that we use the same function everywhere.